### PR TITLE
Refactorization overview

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

File: doc/design/refactorization.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+# Block (in design)
+## the difference with original RNNOp
+- as an operator is more intuitive than `RNNOp`,
+- offers new interface `Eval(targets)` to deduce the minimal block to `Run`,
+- fits the compile-time/ runtime separation design.
+  - during the compilation, `SymbolTable` stores `VarDesc`s and `OpDesc`s and serialize to a `BlockDesc`
+  - when graph executes, a Block with `BlockDesc` passed in creates `Op` and `Var` then `Run`
+
+---
+# Milestone
```
Comment (jacquesqiao) [inline doc/design/refactorization.md]: I think `Milestone` should be in another pr or issue because other parts of this PR is a summary of the refactorized Paddle. They are knowledge but not plan.

File: doc/design/refactorization.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+## Computation Graphs
+
+1. PaddlePaddle represent the computation, training and inference of DL models, by computation graphs.
+
+  1. Please dig into [computation graphs](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/graph.md) for a solid example.
+
+1. Users write Python programs to describe the graphs and run it (locally or remotely).
+
+1. A graph is composed of *variabels* and *operators*.
```
Comment (dzhwinter) [inline doc/design/refactorization.md]: variabels => variables

File: doc/design/refactorization.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    - the MNIST demo needs a Python interface,
+    - the RNN models require the framework to support `LoDTensor`.
+  - determine some timelines,
+  - heavily-relied Ops need to be migrated first,
+  - different models can be migrated parallelly.
+- improve the framework at the same time
+- accept imperfection, concentrated on solving the specific problem at the right price.
+
+---
+# Control the migration quality
```
Comment (dzhwinter) [inline doc/design/refactorization.md]: * Milestone* and *Control the migration quality* is not part of refactorization overview structure,  it's better to move them into the *roadmap* or sth like that.
